codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java

```diff
@@ -213,9 +232,13 @@ public CombinedHttpHeadersImpl setObject(CharSequence name, Iterable<?> values)
             return this;
         }
 
+        private boolean canBeCombined(CharSequence name) {
+            return !nonCombinableHeaders.contains(name);
```

Comment:
If I'm not missing something, this should be case insensitive. You should probably be using the hasher instead.

Reply:
Thats a very good point...  @madgnome please also add a unit test for this.